### PR TITLE
User assignment

### PR DESCRIPTION
**Pull request by Pratyush-Simform**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import MemberPayment from "./components/payments/MemberPayment";
 import CardPayments from "./components/payments/CardPayments";
 import CashPayments from "./components/payments/CashPayments";
 import Layout from "./layout/Layout";
+import Users from "./components/users/Users";
```
Comment (simformniki) [inline src/App.js:9]: Suggestion: You can use lazy loading for all the routes to improve performance.

File: src/components/DndTable/EnhancedTableToolbar.js [hunk 2]
```diff
@@ -96,6 +100,7 @@ let EnhancedTableToolbar = (props) => {
               <IconButton aria-label="Filter list">
                 <Search onClick={() => setSearch(true)} />
                 {title === "Profile Items" ? <AddModal /> : null}
+                {title === "Users" ? <EditUserModal name="ADD" /> : null}
```
Comment (simformniki) [inline src/components/DndTable/EnhancedTableToolbar.js:103]: Store this types of static values in constants.js file.

File: src/components/DndTable/Table.js [hunk 3]
```diff
@@ -625,6 +638,10 @@ class EnhancedTable extends React.Component {
     });
   };
 
+  handleDelete = (row) => {
+    deleteUsers(row.id);
+    window.location.reload();
```
Comment (simformniki) [inline src/components/DndTable/Table.js:643]: why are you using reload() ?

File: src/components/modals/EditUserModal.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      email: row?.email ? row.email : "",
+      first_name: row?.first_name ? row.first_name : "",
+      last_name: row?.last_name ? row.last_name : "",
```
Comment (simformniki) [inline src/components/modals/EditUserModal.js:31]: you can rewrite it as below:
`first_name: row?.first_name  || ""`

File: src/components/modals/EditUserModal.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    },
+    onSubmit: (values) => {
+     if(name === "ADD") {
+         addUsers(values)
+         setOpen(false);
+         window.location.reload();
+     }else {
+         editUsers(row.id, values)
+         setOpen(false);
+         window.location.reload();
```
Comment (simformniki) [inline src/components/modals/EditUserModal.js:42]: Please do not reload. you should just update data source.

File: src/components/modals/EditUserModal.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+     }
+    },
+  });
+  return (
+    <div>
```
Comment (simformniki) [inline src/components/modals/EditUserModal.js:47]: you can wrap with Fragment.

File: src/components/users/Users.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    const [data, setData] = useState([]);
+
+    useEffect(() => {
+    getUsers().then(res => setData(res.data.data.results));
+    }, [])
+
+    return (
+        <div>
+          <EnhancedTable data={data} columnData={cols} name="Users" />
+        </div>
```
Comment (simformniki) [inline src/components/users/Users.js:16]: No need to wrap with <div>

File: src/components/users/Users.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import React, {useState, useEffect} from 'react'
+import { getUsers } from "../../services/userApi"
+import { cols } from "./userColumns"
+import EnhancedTable from "../DndTable/Table"
+
+function Users() {
+    const [data, setData] = useState([]);
+
+    useEffect(() => {
+    getUsers().then(res => setData(res.data.data.results));
```
Comment (simformniki) [inline src/components/users/Users.js:10]: please add catch block to handle errors.

File: src/components/modals/EditUserModal.js [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+import Fade from "@material-ui/core/Fade";
+import EditIcon from "@material-ui/icons/Edit";
+// import Input from "../input/input";
+// import Button from "../input/Button";
+import AddIcon from "@material-ui/icons/Add";
+import { useFormik } from "formik";
+import "../../App.css";
+import { useStyles } from "./styles";
+import TextField from "@material-ui/core/TextField";
+import { addUsers, editUsers } from "../../services/userApi"
```
Comment (simformniki) [inline src/components/modals/EditUserModal.js:13]: Suggestion: You can use absolute path.

File: src/components/modals/EditUserModal.js (hunk 4 above)
Comment (Pratyush-Simform) [reply src/components/modals/EditUserModal.js:31]: done

File: src/components/modals/EditUserModal.js (hunk 6 above)
Comment (Pratyush-Simform) [reply src/components/modals/EditUserModal.js:47]: done

File: src/components/users/Users.js (hunk 7 above)
Comment (Pratyush-Simform) [reply src/components/users/Users.js:16]: done

File: src/components/users/Users.js (hunk 8 above)
Comment (Pratyush-Simform) [reply src/components/users/Users.js:10]: done

File: src/App.js (hunk 1 above)
Comment (Pratyush-Simform) [reply src/App.js:9]: done

File: src/components/DndTable/Table.js (hunk 3 above)
Comment (Pratyush-Simform) [reply src/components/DndTable/Table.js:643]: removed reload

File: src/components/DndTable/EnhancedTableToolbar.js (hunk 2 above)
Comment (Pratyush-Simform) [reply src/components/DndTable/EnhancedTableToolbar.js]: done

File: src/App.js [hunk 10]
```diff
@@ -1,15 +1,17 @@
...
+import Layout from "./layout/Layout"
+const Transaction = React.lazy(() => import('./components/Transaction/transaction'));
+const MemberPayment = React.lazy(() => import('./components/payments/MemberPayment'));
+const CardPayments = React.lazy(() => import('./components/payments/CardPayments'));
+const CashPayments = React.lazy(() => import('./components/payments/CashPayments'));
+const Users = React.lazy(() => import('./components/users/Users'));
+const ProfileItems = React.lazy(() => import('./components/profile/ProfileItems'));
+const Order = React.lazy(() => import('./components/order/Order'));
+const Login = React.lazy(() => import('./components/login/Login'));
+const PaymentProfiles = React.lazy(() => import('./components/profile/PaymentProfiles'));
```
Comment (simformniki) [inline src/App.js:13]: with lazy you need to use <Suspense>.

File: src/components/DndTable/Table.js [hunk 11]
```diff
@@ -625,6 +639,14 @@ class EnhancedTable extends React.Component {
     });
   };
 
+  handleDelete = (row) => {
+    deleteUsers(row.id).then(() =>
+      getUsers().then((res) =>
+        this.setState({ renderer: res.data.data.results })
+      )
+    );
```
Comment (simformniki) [inline src/components/DndTable/Table.js:647]: catch block is missing

Comment (simformniki) [reply src/components/DndTable/Table.js:647]: In catch block, add code to show appropriate message to user.

**Review by simformniki — APPROVED** [review]

**Pull request merged**